File: src/main/java/com/xyomera/register/RegisterItem.java
/*
	Copyright (C) 2026 xyomera,
	All rights reserved.
	but based on the Apache 2.0 license.
 */

package com.xyomera.register;

import java.util.function.Function;

import net.minecraft.core.Registry;
import net.minecraft.core.registries.BuiltInRegistries;
import net.minecraft.core.registries.Registries;
import net.minecraft.resources.Identifier;
import net.minecraft.resources.ResourceKey;
import net.minecraft.world.item.Item;

public class RegisterItem {
    // Create Resource Key
    public static ResourceKey<Item> create(String name) {
        return ResourceKey.create(Registries.ITEM, Identifier.fromNamespaceAndPath("supereats", name));
    }

    // Register Item
    public static Item register(ResourceKey<Item> itemKey, Function<Item.Properties, Item> itemFactory, Item.Properties settings) {
        Item item = itemFactory.apply(settings.setId(itemKey));

        Registry.register(BuiltInRegistries.ITEM, itemKey, item);

        return item;
    }
}

File: src/main/kotlin/com/xyomera/register/RegisterItem.kt
/*
	Copyright (C) 2026 xyomera,
	All rights reserved.
	but based on the Apache 2.0 license.
 */

package com.xyomera.register

import net.minecraft.core.Registry
import net.minecraft.core.registries.BuiltInRegistries;
import net.minecraft.core.registries.Registries
import net.minecraft.resources.Identifier;
import net.minecraft.resources.ResourceKey
import net.minecraft.world.item.Item

object RegisterItem {
    // Create Resource Key
    @JvmStatic
    fun create(name: String) : ResourceKey<Item> {
        return ResourceKey.create(Registries.ITEM, Identifier.fromNamespaceAndPath("supereats", name))
    }

    // Register Item
    @JvmStatic
    fun register(itemKey: ResourceKey<Item>, itemFactory: (Item.Properties) -> Item, settings: Item.Properties) : Item {
        val props: Item.Properties = settings.setId(itemKey)
        val item: Item = itemFactory(props)
        Registry.register(BuiltInRegistries.ITEM, itemKey, item)
        return item
    }
}